pkg/registry/apis/dashboard/legacy_storage.go

```diff
@@ -44,9 +47,61 @@ func (s *DashboardStorage) NewStore(scheme *runtime.Scheme, defaultOptsGetter ge
 	if err != nil {
 		return nil, err
 	}
-	client := resource.NewLocalResourceClient(server)
+	client := resource.NewDirectResourceClient(server) // same context
```

Comment:
🤯 so... the clever plan to simplify dashboard SQL <> apiserver logic by re-using a simplified instance of unified storage sort of bites us in the butt here.

Reusing that logic also means we talk to the *local* SQL via a fake grpc connection that does not use the same context.  This PR modifies our connection so we use a direct connection